pkg/testing/table.go

```diff
@@ -0,0 +1,281 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package testing
+
+import (
+	"bytes"
+	"context"
+	"path"
+	"reflect"
+	"strings"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/knative/pkg/kmeta"
+	kntesting "github.com/knative/pkg/reconciler/testing"
+	"github.com/projectriff/riff/pkg/riff"
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/runtime"
+	clientgotesting "k8s.io/client-go/testing"
+)
+
+type CommandTable []CommandTableRecord
+
+type CommandTableRecord struct {
```

Comment:
It's not immediately obvious how to use this struct properly, so developers might be reduced to copying and hacking other tests without necessarily understanding any subtleties. Compare this with ginkgo/gomega which has good documentation and is familiar to most of the riff team. If we decide to switch riff to this struct instead of ginkgo, I think we'll need documentation for developers.

Reply:
Yes, it needs docs. I've been avoiding docs while things are in flux.